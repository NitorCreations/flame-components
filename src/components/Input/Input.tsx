import * as React from 'react';

interface IInputProps {
  placeholder?: string;
  onChange?: () => void;
  disabled?: boolean;
  className?: string;
}

export default class Input extends React.PureComponent<IInputProps, {}> {
  public render() {
    const { onChange, placeholder, className, disabled } = this.props;

    return (
      <input
        className={`flame-input ${className ? className : ''}`}
        placeholder={placeholder}
        disabled={disabled}
        onChange={onChange}
      />
    );
  }
}