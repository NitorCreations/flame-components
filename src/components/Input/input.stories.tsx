import * as React from 'react';
import { storiesOf } from '@storybook/react';
import Input from './Input';

storiesOf('Input', module)
  .add('Normal', () => (
      <Input placeholder="input" />
    )
  )
  .add('Disabled', () => (
      <Input placeholder="input" disabled />
    )
  );