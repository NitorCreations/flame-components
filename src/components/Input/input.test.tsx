import * as React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input', () => {
  it('Should render correctly', () => {
    const component = shallow(<Input />);

    expect(component).toMatchSnapshot();
  });

  it('Should be correctly enabled', () => {
    const component = shallow(<Input disabled={false} />);

    expect(component).toMatchSnapshot();
  });

  it('Should be correctly disabled', () => {
    const component = shallow(<Input disabled />);

    expect(component).toMatchSnapshot();
  });
});