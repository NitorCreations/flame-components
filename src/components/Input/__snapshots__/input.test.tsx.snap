// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should be correctly disabled 1`] = `
<input
  className="flame-input "
  disabled={true}
/>
`;

exports[`Input Should be correctly enabled 1`] = `
<input
  className="flame-input "
  disabled={false}
/>
`;

exports[`Input Should render correctly 1`] = `
<input
  className="flame-input "
/>
`;
